native-packages/fs-search/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+extern crate napi;
+#[macro_use]
+extern crate napi_derive;
+
+use std::convert::TryInto;
+use napi::{CallContext, JsNumber, JsString, JsNull, Either, JsObject, Result};
+use std::path::Path;
+
+#[js_function(2)]
+fn find_ancestor_file(ctx: CallContext) -> Result<Either<JsNull, JsString>> {
+  let names = ctx.get::<JsObject>(0)?;
+  let length: u32 = names.get_named_property::<JsNumber>("length")?.try_into()?;
+  let mut filenames = Vec::new();
+  for i in 0..length {
+    filenames.push(names.get_element::<JsString>(i)?.into_utf8()?);
+  }
+
+  let f = ctx.get::<JsString>(1)?.into_utf8()?;
+  let from = Path::new(f.as_str()?);
+
+  for dir in from.ancestors() {
+    // Break if we hit a node_modules directory
+    if let Some(filename) = dir.file_name() {
```

Comment:
Shouldn't this also stop once the project root is reached?

Reply:
Yeah probably. Doesn't look like it did before though :/

Reply:
Which also explains this terserrc is applied:
```
.
├── .terserrc
└── project
    ├── dist
    │   ├── index.js
    │   └── index.js.map
    ├── index.js
    ├── package.json
    └── yarn.lock
```
```json
{
  "compress": false,
  "mangle": false
}
```

Reply:
Let's do this as a separate task since it's not a regression and looks like a bunch of code will have to be changed.